File: core/src/main/java/io/github/manami/core/commands/CmdChangeLocation.java
package io.github.manami.core.commands;

import io.github.manami.core.Manami;
import io.github.manami.dto.entities.Anime;

/**
 * Command for changing the location.
 */
public class CmdChangeLocation extends AbstractReversibleCommand {

  /**
   * Constructor.
   *
   * @param anime Anime to change.
   * @param newValue The new value.
   * @param application Instance of the application which reveals access to the persistence functionality.
   */
  public CmdChangeLocation(final Anime anime, final String newValue, final Manami application) {
    app = application;
    oldAnime = anime;
    newAnime = new Anime(
        oldAnime.getTitle(),
        oldAnime.getInfoLink(),
        oldAnime.getEpisodes(),
        oldAnime.getType(),
        oldAnime.getLocation(),
        oldAnime.getThumbnail(),
        oldAnime.getPicture(),
        oldAnime.getId()
    );
    newAnime.setLocation(newValue);
  }
}

File: core/src/main/java/io/github/manami/core/commands/CmdChangeLocation.kt
package io.github.manami.core.commands

import io.github.manami.dto.entities.Anime
import io.github.manami.persistence.Persistence


/**
 * Command for changing the location.
 *
 * @param anime Anime to change.
 * @param newValue The new value.
 * @param persistence
 */
internal class CmdChangeLocation(
        private val anime: Anime,
        private val newValue: String,
        private val persistence: Persistence
) : AbstractReversibleCommand(persistence) {

    init {
        oldAnime = anime
        newAnime = oldAnime?.copy(location = newValue)
    }
}
